project.go

```diff
@@ -64,3 +65,28 @@ func (p *Project) MakeParams() gps.SolveParameters {
 
 	return params
 }
+
+// BackupVendor looks for existing vendor directory and if it's not empty,
+// creates a backup of it to a new directory with the provided suffix.
+func BackupVendor(vpath, suffix string) (string, error) {
+	// Check if there's a non-empty vendor directory
+	vendorExists, err := IsNonEmptyDir(vpath)
+	if err != nil {
+		return "", err
+	}
+	if vendorExists {
+		vendorbak := vpath + "-" + suffix
+		// Check if a directory with same name exists
+		if _, err = os.Stat(vendorbak); os.IsNotExist(err) {
+			// Copy existing vendor to vendor-{suffix}
+			if err := CopyDir(vpath, vendorbak); err != nil {
```

Comment:
@sdboyer now that I think of it again, renaming makes more sense 😅 no point copying and leaving the file when it would be overwritten in the next phase of init. Will change it.